packages/bpk-component-modal/src/BpkModalDialog.js

```diff
@@ -95,15 +94,15 @@ const BpkModalDialog = (props: Props) => {
         id={props.id}
         tabIndex="-1"
         role="dialog"
-        aria-labelledby={headingId}
+        aria-labelledby={props.showHeader ? headingId : null}
         className={classNames.join(' ')}
         ref={props.dialogRef}
       >
         {props.INTERNAL__outerComponent}
         {props.showHeader && (
           <header className={getClassName('bpk-modal__header')}>
             <BpkNavigationBar
-              id={navigationId}
+              id={headingId}
```

Comment:
We pass `headingId` instead of `navigationId` as the `id` prop is purely used for setting `aria-labelledby` in the Navigation component so it ensures the navbar can also target the correct title component